### PR TITLE
WIP: Implement geofence gRPC service

### PR DESCRIPTION
**Pull request by coderkalyan**
Hi, I have been working on the service implementation for Geofence in gRPC. This isn't complete - I am having some errors. The following was produced after I compiled MAVSDK-Python with run_protoc.sh, built `mavsdk_server` using the docs provided on the website, copied it to `mavsdk/bin` and installed with pip.
1) Error: I can successfully create `Polygon` and `Point` objects in mavsdk. However, when I call `mavsdk.geofence.upload_geofence(...)`, theres an error saying it's unimplemented:
```-- Uploading geofence
Task exception was never retrieved
future: <Task finished coro=<run() done, defined at mission.py:11> exception=<_Rendezvous of RPC that terminated with:
        status = StatusCode.UNIMPLEMENTED
        details = ""
        debug_error_string = "{"created":"@1563564989.499149902","description":"Error received
from peer ipv4:127.0.0.1:50051","file":"src/core/lib/surface/call.cc","file_line":1052,"grpc_message":"","grpc_status":12}"
>>
Traceback (most recent call last):
  File "mission.py", line 53, in run
    await drone.geofence.upload_geofence([pgn])
  File "/home/kalyan/src/MAVSDK-Python/mavsdk/plugins/geofence.py", line 311, in upload_geofence
    response = await self._stub.UploadGeofence(request)
  File "/home/kalyan/src/MAVSDK-Python/venv/lib/python3.7/site-packages/aiogrpc/channel.py", line 40, in __call__
    return await fut
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with:
        status = StatusCode.UNIMPLEMENTED
        details = ""
        debug_error_string = "{"created":"@1563564989.499149902","description":"Error received
from peer ipv4:127.0.0.1:50051","file":"src/core/lib/surface/call.cc","file_line":1052,"grpc_message":"","grpc_status":12}"
>
```
What is causing this error?
2) I would like some feedback on how my code is, since it's my first time working with gRPC!

### CODE REVIEW

**Comment by JonasVautherin** [comment]
You also need to instantiate your plugin in [grpc_server](https://github.com/mavlink/MAVSDK/blob/develop/src/backend/src/grpc_server.cpp#L19) :blush:. That's probably why you get this `UNIMPLEMENTED` error.

File: src/backend/src/plugins/geofence/geofence_service_impl.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    }
+
+                    result_promise.set_value();
+            });
+    }
+
+    rpc::geofence::GeofenceResult*
+    generateRPCGeofenceResult(const mavsdk::Geofence::Result result) const
+    {
+        auto rpc_result = static_cast<rpc::geofence::GeofenceResult::Result>(result);
```
Comment (JonasVautherin) [inline src/backend/src/plugins/geofence/geofence_service_impl.h]: This works, and that's how we do it everywhere, but that's super unsafe. Hence the unit tests (e.g. [here](https://github.com/mavlink/MAVSDK/blob/develop/src/backend/test/action_service_impl_test.cpp#L96)).

**Comment by JonasVautherin** [comment]
Seems like we're almost there! I see that github says that you have "conflicting files". Dont forget to rebase `develop` :slightly_smiling_face:

**Comment by coderkalyan** [comment]
I have accidentally caused some very bad build issues by rebasing, so I've decided to close this PR and start again.

**Comment by JonasVautherin** [comment]
You can always `push -f` on the same branch, and that will update the PR. But opening a new PR is fine as well :+1:.

**Comment by JonasVautherin** [comment]
Any news on that front? :-)

**Comment by coderkalyan** [comment]
Working on it :)

**Pull request closed without merging**